openshift/backup/backup-deploy.yaml

```diff
@@ -0,0 +1,447 @@
+kind: Template
+apiVersion: v1
+metadata:
+  name: ${NAME}-deployment-template
+objects:
+  - kind: NetworkPolicy
+    apiVersion: networking.k8s.io/v1
+    metadata:
+      name: ${NAME}
+      labels:
+        name: ${NAME}
+        app: ${APP_NAME}
+        env: ${TAG_NAME}
+    spec:
+      description: |
+        Allow the ${NAME} to access databases in this namespace.
+      ingress:
+        - from:
+            - podSelector:
+                matchLabels:
+                  name: ${NAME}
+                  app: ${APP_NAME}
+                  role: ${ROLE}
+                  env: ${TAG_NAME}
+              namespaceSelector:
+                matchLabels:
+                  name: ${NAMESPACE_NAME}
+                  environment: ${TAG_NAME}
+      podSelector:
+        matchLabels:
+          backup: "true"
+          env: ${TAG_NAME}
+
+  - kind: PersistentVolumeClaim
+    apiVersion: v1
+    metadata:
+      name: ${BACKUP_VOLUME_NAME}
+      labels:
+        name: ${NAME}
+        app: ${APP_NAME}
+        role: ${ROLE}
+        env: ${TAG_NAME}
+    spec:
+      storageClassName: ${BACKUP_VOLUME_CLASS}
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: ${BACKUP_VOLUME_SIZE}
+
+  - kind: PersistentVolumeClaim
+    apiVersion: v1
+    metadata:
+      name: ${VERIFICATION_VOLUME_NAME}
+      labels:
+        name: ${NAME}
+        app: ${APP_NAME}
+        role: ${ROLE}
+        env: ${TAG_NAME}
+    spec:
+      storageClassName: ${VERIFICATION_VOLUME_CLASS}
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: ${VERIFICATION_VOLUME_SIZE}
+
+  - kind: Secret
+    apiVersion: v1
+    metadata:
+      name: ${NAME}
+      labels:
+        name: ${NAME}
+        app: ${APP_NAME}
+        role: ${ROLE}
+        env: ${TAG_NAME}
+    type: Opaque
+    stringData:
+      webhook-url: ${WEBHOOK_URL}
+      webhook-url-host: ${WEBHOOK_URL_HOST}
+
+  - kind: Secret
+    apiVersion: v1
+    metadata:
+      name: ${FTP_SECRET_KEY}
+      labels:
+        name: ${NAME}
+        app: ${APP_NAME}
+        role: ${ROLE}
+        env: ${TAG_NAME}
+    type: Opaque
+    stringData:
+      ftp-url: ${FTP_URL}
+      ftp-user: ${FTP_USER}
+      ftp-password: ${FTP_PASSWORD}
+      ftp-url-host: ${FTP_URL_HOST}
```

Comment:
when I worked on backup containers, I don't recall ftp usage. Are we transferring the backup to a different location, and not the PV ?

Reply:
FTP usage is a flag that is only used if a URL is present. I think if you read the first part of the readme here https://github.com/BCDevOps/backup-container you'll see that's the case?